src/apppermission.h

```diff
@@ -0,0 +1,54 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef APPPERMISSION_H
+#define APPPERMISSION_H
+
+#include <QObject>
+#include <QAbstractListModel>
+#include "applistprovider.h"
+
+class AppPermission : public QAbstractListModel
+{
+    Q_OBJECT
+    Q_DISABLE_COPY_MOVE(AppPermission)
+    Q_PROPERTY(bool listReady READ listReady NOTIFY readyChanged)
```

Comment:
Why do you need this? What does it happen without it?

Reply:
maybe we can show a cursor when we retrieve the list of apps. But in this case, `m_appList.isEmpty()` is enough.

Reply:
I'm currently showing a spinner while fetching :)